driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -248,11 +249,18 @@ public final class MongoCredential {
      * {@code "*.mongodb.net", "*.mongodb-qa.net", "*.mongodb-dev.net", "*.mongodbgov.net", "localhost", "127.0.0.1", "::1"}
      *
      * @see #createOidcCredential(String)
-     * @since 4.10
+     * @since 5.1
      */
     public static final List<String> DEFAULT_ALLOWED_HOSTS = Collections.unmodifiableList(Arrays.asList(
             "*.mongodb.net", "*.mongodb-qa.net", "*.mongodb-dev.net", "*.mongodbgov.net", "localhost", "127.0.0.1", "::1"));
 
+    /**
+     * The token resource.
+     *
+     * @since 5.1
+     */
+    public static final String TOKEN_RESOURCE_KEY = "TOKEN_RESOURCE";
```

Comment:
I am not sure about specifying the environments explicitly, since it seems easy to forget to update these, and a user configuring (for example) GCP should know that their environment needs it.